### PR TITLE
discord: discord : found a required dependency

### PR DESCRIPTION
**Pull request by tororutsu**
#### Testing the changes
- I tested the changes in this PR: **YES**
- 
After building Discord I would get a message saying my install was corrupt. i saw this [post](https://www.reddit.com/r/voidlinux/comments/11omnrp/comment/jcbev31/?utm_source=share&utm_medium=web2x&context=3) and worked to find which package i needed

### CODE REVIEW

**Comment by classabbyamp** [comment]
I cannot reproduce this, with an extremely minimal environment:

1. booted the base live iso
2. updated `xbps`
3. installed `xorg xterm discord`
4. did `startx`
5. launched discord from one of the xterms there

<details><summary>installed packages (<code>xbps-query -l | cut -d' ' -f 2</code>) </summary>
<p>

```
acl-2.3.1_1
acpid-2.0.33_2
adwaita-icon-theme-43_1
alsa-lib-1.2.8_1
at-spi2-atk-2.38.0_1
at-spi2-core-2.44.0_1
atk-2.36.0_1
attr-2.5.1_1
avahi-libs-0.8_2
base-files-0.142_14
base-system-0.114_1
bash-5.1.016_2
brotli-1.0.9_2
btrfs-progs-5.19.1_1
bzip2-1.0.8_1
c-ares-1.18.1_1
ca-certificates-20211016+3.83_1
cairo-1.16.0_2
coreutils-8.32_4
cpio-2.13_1
cryptsetup-2.4.3_1
curl-7.87.0_2
dash-0.5.11.5_2
dbus-1.14.4_1
dbus-glib-0.112_1
dbus-libs-1.14.4_1
dbus-x11-1.14.4_1
dejavu-fonts-ttf-2.37_2
device-mapper-2.02.187_2
dhcpcd-9.4.1_2
dialog-1.3.20220526_1
diffutils-3.8_1
discord-0.0.25_1
dnssec-anchors-20181012_2
dosfstools-4.2_1
dracut-053_4
e2fsprogs-1.46.5_1
e2fsprogs-libs-1.46.5_1
efibootmgr-18_1
encodings-1.0.6_1
ethtool-5.16_1
eudev-3.2.11_1
eudev-libudev-3.2.11_1
expat-2.4.9_1
f2fs-tools-1.15.0_1
file-5.42_2
findutils-4.9.0_1
font-adobe-100dpi-1.0.3_7
font-adobe-75dpi-1.0.3_7
font-adobe-utopia-100dpi-1.0.4_7
font-adobe-utopia-75dpi-1.0.4_7
font-adobe-utopia-type1-1.0.4_7
font-alias-1.0.4_2
font-bh-100dpi-1.0.3_6
font-bh-75dpi-1.0.3_6
font-bh-lucidatypewriter-100dpi-1.0.3_6
font-bh-lucidatypewriter-75dpi-1.0.3_6
font-bh-ttf-1.0.3_6
font-bh-type1-1.0.3_6
font-bitstream-100dpi-1.0.3_6
font-bitstream-75dpi-1.0.3_6
font-bitstream-type1-1.0.3_6
font-cursor-misc-1.0.3_6
font-daewoo-misc-1.0.3_6
font-dec-misc-1.0.3_6
font-ibm-type1-1.0.3_6
font-isas-misc-1.0.3_6
font-jis-misc-1.0.3_5
font-misc-misc-1.1.2_7
font-mutt-misc-1.0.3_6
font-util-1.4.0_1
fontconfig-2.14.2_1
freetype-2.12.1_2
fribidi-1.0.12_1
fuse-2.9.9_1
gawk-5.1.1_1
gdk-pixbuf-2.42.6_1
glib-2.74.0_3
glibc-2.36_1
glibc-locales-2.32_2
gmp-6.2.1_1
graphite-1.3.14_1
grep-3.8_1
grub-2.06_2
grub-i386-efi-2.06_2
grub-x86_64-efi-2.06_2
gtk+3-3.24.37_1
gtk-update-icon-cache-2.24.33_2
gzip-1.12_2
hwids-0.361_1
iana-etc-20220601_1
iceauth-1.0.9_1
icu-libs-71.1_1
inih-53_1
iproute2-5.19.0_1
iptables-1.8.7_1
iputils-20211215_2
ipw2100-firmware-1.3_6
ipw2200-firmware-3.1_6
iw-5.19_1
jansson-2.14_1
jbigkit-libs-2.2_2
json-c-0.16_1
kbd-2.5.1_1
kmod-30_1
kpartx-0.9.1_1
lcms2-2.14_1
less-608_1
libICE-1.1.1_1
libSM-1.2.3_1
libX11-1.8.4_1
libXau-1.0.11_1
libXaw-1.0.14_1
libXcomposite-0.4.6_1
libXcursor-1.2.1_2
libXdamage-1.1.6_1
libXdmcp-1.1.4_1
libXext-1.3.5_1
libXfixes-6.0.0_1
libXfont2-2.0.6_1
libXfontcache-1.0.5_2
libXft-2.3.7_1
libXi-1.8_1
libXinerama-1.1.5_1
libXmu-1.1.4_1
libXpm-3.5.14_1
libXrandr-1.5.3_1
libXrender-0.9.11_1
libXt-1.2.1_1
libXtst-1.2.4_1
libXv-1.0.12_1
libXvMC-1.0.13_1
libXxf86misc-1.0.4_1
libXxf86vm-1.1.5_1
libaio-0.3.112_1
libarchive-3.6.1_1
libargon2-20190702_3
libatomic-12.2.0_2
libblkid-2.38.1_1
libcap-2.54_1
libcap-ng-0.8.2_2
libcap-progs-2.54_1
libcolord-1.4.6_2
libcrypto1.1-1.1.1q_1
libcryptsetup-2.4.3_1
libcups-2.4.2_3
libcurl-7.87.0_2
libcxx-12.0.1_4
libdatrie-0.2.13_1
libdb-5.3.28_8
libdrm-2.4.114_1
libedit-20210910.3.1_1
libefivar-37_2
libelf-0.187_1
libepoxy-1.5.10_1
libev-4.33_1
libevdev-1.13.0_1
libfdisk-2.38.1_1
libffi-3.3_2
libfontenc-1.1.4_1
libgbm-22.3.5_1
libgcc-10.2.1pre1_3
libglapi-22.3.5_1
libglvnd-1.4.0_1
libgudev-237_1
libgusb-0.3.10_1
libharfbuzz-5.1.0_2
libinput-1.22.1_1
libjpeg-turbo-2.1.4_1
libkmod-30_1
libldap-2.4.58_1
libldns-1.8.3_1
libllvm12-12.0.1_4
liblz4-1.9.3_1
liblzma-5.2.6_1
libmagic-5.42_2
libmcpp-2.7.2_8
libmnl-1.0.5_1
libmount-2.38.1_1
libnetfilter_conntrack-1.0.8_1
libnfnetlink-1.0.1_4
libnftnl-1.2.3_1
libnl3-3.5.0_1
libnotify-0.8.1_1
libpcap-1.10.1_1
libpciaccess-0.16_1
libpcre2-10.39_2
libpng-1.6.39_1
libreadline8-8.1.000_1
librsvg-2.52.2_1
libsasl-2.1.27_2
libsensors-3.6.0_1
libsmartcols-2.38.1_1
libssh-0.9.6_1
libssh2-1.10.0_1
libssl1.1-1.1.1q_1
libstdc++-10.2.1pre1_3
libthai-0.1.29_1
libtirpc-1.3.2_1
liburcu-0.12.2_1
libusb-1.0.26_1
libutempter-1.2.1_2
libuuid-2.38.1_1
libwacom-2.6.0_1
libxatracker-22.3.5_1
libxbps-0.59.1_8
libxcb-1.14_2
libxcvt-0.1.1_1
libxkbcommon-1.5.0_1
libxkbfile-1.1.0_1
libxml2-2.9.13_4
libxshmfence-1.3_2
libzstd-1.5.2_1
linux-5.19_1
linux-base-2021.07.21_1
linux-firmware-amd-20220411_1
linux-firmware-broadcom-20220411_1
linux-firmware-intel-20220411_1
linux-firmware-network-20220411_1
linux-firmware-nvidia-20220411_1
linux5.19-5.19.10_1
lowdown-1.0.0_1
lvm2-2.02.187_2
lzo-2.10_2
man-pages-5.13_2
mcpp-2.7.2_8
mdadm-4.2_1
mdocml-1.14.6_6
mesa-22.3.5_1
mesa-dri-22.3.5_1
mit-krb5-libs-1.20.1_1
mkfontscale-1.2.2_1
mtdev-1.1.6_1
ncurses-6.3_2
ncurses-base-6.3_2
ncurses-libs-6.3_2
nettle-3.8_1
nghttp2-1.51.0_1
nspr-4.35_1
nss-3.89_1
nvi-1.81.6_19
oclock-1.0.5_1
openssh-9.0p1_1
openssl-1.1.1q_1
os-prober-1.79_1
pam-1.5.2_2
pam-base-0.4_2
pam-libs-1.5.2_2
pango-1.50.14_1
pciutils-3.7.0_1
pick-4.0.0_1
pixman-0.42.2_1
popt-1.18_1
procps-ng-3.3.17_2
removed-packages-0.1_73
run-parts-4.11.2_1
runit-2.1.2_11
runit-void-20220329_1
sed-4.8_1
sessreg-1.1.2_1
setxkbmap-1.3.3_1
shadow-4.8.1_2
shared-mime-info-2.2_1
smproxy-1.0.6_2
sqlite-3.41.1_1
sudo-1.9.11p3_1
tar-1.34_2
thin-provisioning-tools-0.9.0_1
tiff-4.4.0_1
traceroute-2.1.0_4
transset-1.0.3_1
tzdata-2022d_1
usbutils-014_2
util-linux-2.38.1_1
util-linux-common-2.38.1_1
void-artwork-20220303_1
void-docs-2021.10.06_1
void-docs-browse-2021.10.06_1
wayland-1.21.0_1
webrtc-audio-processing-0.3.1_1
which-2.21_3
wifi-firmware-1.3_4
wpa_supplicant-2.10_1
x11perf-1.6.1_1
xauth-1.1.2_1
xbacklight-1.2.3_1
xbps-0.59.1_8
xbps-triggers-0.124_1
xcalc-1.1.1_1
xcb-util-0.4.1_1
xclipboard-1.1.4_1
xclock-1.1.1_1
xcmsdb-1.0.6_1
xconsole-1.0.8_1
xcursorgen-1.0.7_1
xdg-utils-1.1.3_5
xdpyinfo-1.3.3_1
xev-1.2.5_1
xeyes-1.2.0_1
xf86-input-evdev-2.10.6_2
xf86-input-libinput-1.2.1_1
xf86-input-synaptics-1.9.1_1
xf86-input-vmmouse-13.1.0_4
xf86-input-wacom-1.1.0_1
xf86-video-amdgpu-22.0.0_1
xf86-video-ati-19.1.0_4
xf86-video-dummy-0.4.0_1
xf86-video-fbdev-0.5.0_2
xf86-video-intel-2.99.917.20210115_2
xf86-video-nouveau-1.0.17_2
xf86-video-vesa-2.5.0_2
xf86-video-vmware-13.3.0_4
xfontsel-1.1.0_1
xfsprogs-5.18.0_1
xgamma-1.0.7_1
xhost-1.0.9_1
xinit-1.4.2_1
xinput-1.6.3_1
xkbcomp-1.4.6_1
xkbevd-1.1.4_2
xkbutils-1.0.4_2
xkeyboard-config-2.38_1
xkill-1.0.6_1
xload-1.1.4_1
xlogo-1.0.6_1
xlsatoms-1.1.3_2
xlsclients-1.1.5_1
xmessage-1.0.6_1
xmodmap-1.0.11_1
xorg-7.6_6
xorg-apps-7.6_9
xorg-fonts-7.6_5
xorg-input-drivers-7.6_4
xorg-server-21.1.7_1
xorg-server-common-21.1.7_1
xorg-video-drivers-7.6_22
xpr-1.1.0_1
xprop-1.2.6_1
xrandr-1.5.2_1
xrdb-1.2.1_1
xrefresh-1.0.6_1
xset-1.2.5_1
xsetroot-1.1.3_1
xterm-379_1
xtools-minimal-0.65_1
xvinfo-1.1.4_2
xwd-1.0.8_1
xwininfo-1.1.5_1
xwud-1.0.5_1
zd1211-firmware-1.5_3
zlib-1.2.12_4
``` 

</p>
</details>

**Comment by tororutsu** [comment]
> 

Hmm very interesting. I'll take a further look. Maybe do a fresh install and install discord first

**Comment by paper42** [comment]
Someone on Reddit reported a similar issue and that installing libreoffice fixed it for them. Maybe there is a common dependency of lo and gst-plugins-bad that is necessary for this to work?

**Comment by tororutsu** [comment]
The problem is @classabbyamp was able to build on a base iso. So that leaves me to believe something else is broke

**Comment by tororutsu** [comment]
> Someone on Reddit reported a similar issue and that installing libreoffice fixed it for them. Maybe there is a common dependency of lo and gst-plugins-bad that is necessary for this to work?

I also referenced this in OP, that's how I found out lol

**Comment by Clos3y** [comment]
@tororutsu what WM/DE are you using just out of curiosity? Just curious if Wayland/xorg will play any part.

**Comment by tororutsu** [comment]
@Clos3y I am using Herbstluftwm, and just installed the whole xorg package

**Comment by tororutsu** [comment]
@classabbyamp I have just done a base install and followed your steps, and can confirm that Discord will after login show a corrupted install. I have all my steps written down and package logs. I just have no idea how to upload that stuff on a super minimal install with default twm. I couldn't even close discord lol

**Comment by classabbyamp** [comment]
doing it from the base iso is ~identical to a base install

**Comment by tororutsu** [comment]
@classabbyamp how do we move forward from here. The problem is confirmed, I will run some more tests when I am able, but at the bare minimum, that plugin is required or at least some of that plugins dependencies are

File: srcpkgs/discord/template [hunk 1]
```diff
@@ -1,10 +1,10 @@
 # Template file for 'discord'
 pkgname=discord
 version=0.0.25
-revision=1
+revision=2
 archs="x86_64"
 depends="alsa-lib dbus-glib gtk+3 libnotify nss libXtst libcxx libatomic
- xdg-utils webrtc-audio-processing"
+ xdg-utils webrtc-audio-processing gst-plugins-bad1"
```
Comment (classabbyamp) [inline srcpkgs/discord/template:7]:
```suggestion
 xdg-utils webrtc-audio-processing libXScrnSaver"
```
I went through all the dependencies of `gst-plugins-bad1`, and the missing one is `libXScrnSaver`.

further confirmed with `strace`:
```
$ strace -Z -ff -o discord.log Discord
$ grep libXss discord.log.* | wc -l
384
$ sudo xbps-install libXScrnSaver
$ strace -Z -ff -o discord-xss.log Discord
$ grep libXss discord-xss.log.* | wc -l
1
```

**Review by classabbyamp — CHANGES_REQUESTED** [review]
commit message should be something like `discord: add missing dependency`

**Comment by classabbyamp** [comment]
please squash the commits then it should be good

**Comment by tororutsu** [comment]
> please squash the commits then it should be good
Srry noob here, how do i do that?

**Comment by classabbyamp** [comment]
if you made this commit on the web, you'll need to `git pull`.

then `git rebase -i master` will open an editor with a todo list. just delete the line with the old commit, then save and close the file. after it finishes successfully, `git push --force`

**Comment by classabbyamp** [comment]
if you made the second commit on github, `git pull`

then `git rebase -i master`. this will open an editor. on the line with the second commit, change `pick` to `f -C` so it squashes it into the first commit, but only keeps the commit message of the second. then save & close the file

if it succeeds and looks right in `git log` and `git show`, `git push --force`
